api/resources_portal/views/organization_grant.py

```diff
@@ -0,0 +1,68 @@
+from rest_framework import viewsets
+from rest_framework.exceptions import MethodNotAllowed
+from rest_framework.permissions import BasePermission, IsAuthenticated
+from rest_framework.response import Response
+
+from rest_framework_extensions.mixins import NestedViewSetMixin
+
+from resources_portal.models import Grant, GrantOrganizationAssociation, Organization
+from resources_portal.views.relation_serializers import GrantRelationSerializer
+
+
+class IsMemberOfOrganization(BasePermission):
+    def has_permission(self, request, view):
+        organization = Organization.objects.get(pk=view.kwargs["parent_lookup_organizations"])
+
+        return request.user in organization.members.all()
+
+
+class OwnsGrantAndOrganization(BasePermission):
+    def has_permission(self, request, view):
+        if view.action == "create":
+            grant = Grant.objects.get(pk=request.data["id"])
+            organization = Organization.objects.get(pk=view.kwargs["parent_lookup_organizations"])
+        else:
+            grant = Grant.objects.get(pk=view.kwargs["pk"])
+            organization = Organization.objects.get(pk=view.kwargs["parent_lookup_organizations"])
```

Comment:
This line is the same in both conditions, it can be moved down so we only change the way we retrieve the grant.

Reply:
👍